oauthproxy.go

```diff
@@ -792,6 +801,11 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	switch err {
 	case nil:
+		if len(p.AllowedGroups) > 0 && !p.validateGroups(session.Groups) {
+			p.ErrorJSON(rw, http.StatusUnauthorized)
```

Comment:
If you look up a bit in this file, you can see an example of using `p.ErrorPage` + a `logger.PrintAuthf` -- I think following that pattern would be appropriate here.

Reply:
Done!